src/Text/Pandoc/Builder.hs

```diff
@@ -566,6 +569,28 @@ simpleCaption = caption Nothing
 emptyCaption :: Caption
 emptyCaption = simpleCaption mempty
 
+-- | Bidirectional patter synonym
+--
+-- It can pass as a Block constructor
+--
+-- λ> SimpleFigure nullAttr [] (T.pack "") (T.pack "")
+-- Para [Image ("",[],[]) [] ("","fig:")]
+--
+-- It can be used to pattern match
+-- foo :: Block -> ...
+-- foo (SimpleFigure attr figureCaption src title) -> handleFigure ...
+pattern SimpleFigure :: Attr -> [Inline] -> Text -> Text -> Block
+pattern SimpleFigure attr figureCaption src title <- Para [Image attr figureCaption (src, title)] where
```

Comment:
Nice! We should also check that title has a `fig:` prefix, which we can do with this view pattern:
``` haskell
Image attr figureCaption (src, (T.stripPrefix "fig:" -> Just title))
```